packages/jest-runtime/src/index.ts

```diff
@@ -553,6 +553,69 @@ export default class Runtime {
       return globals;
     }
 
+    if (specifier.startsWith('data:')) {
+      if (
+        this._shouldMock(
+          referencingIdentifier,
+          specifier,
+          this._explicitShouldMockModule,
+          {conditions: this.esmConditions},
+        )
+      ) {
+        return this.importMock(referencingIdentifier, specifier, context);
+      }
+
+      const fromCache = this._esmoduleRegistry.get(specifier);
+
+      if (fromCache) {
+        return fromCache;
+      }
+
+      const match = specifier.match(
+        /^data:text\/javascript;(charset=utf-8|base64),(.*)$/,
```

Comment:
looking at https://nodejs.org/api/esm.html#data-imports we should support JSON as well (plus wasm, but I think we can just throw if that is provided)